server/apiv2/handlers/stores.go

```diff
@@ -0,0 +1,200 @@
+		c.IndentedJSON(http.StatusOK, storeInfo)
+	}
+}
+
+// DeleteStoreByID will delete the store according to the given ID.
+// @Tags stores
+// @version 2.0
+// @Summary Delete a store's information.
+// @Param id path integer true "Store Id"
+// @Produce json
+// @Success 200
+// @Failure 400 {string} string "The input is invalid."
+// @Router /stores [delete]
+func DeleteStoreByID() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		rc := c.MustGet(apiutil.ClusterKey).(*cluster.RaftCluster)
+		idParam := c.Param(StoreIDParamKey)
+		id, err := strconv.ParseUint(idParam, 10, 64)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrStrconvParseUint.Wrap(err).FastGenWithCause().Error())
+			return
+		}
+
+		var force bool
+		forceQuery, exist := c.GetQuery("force")
+		if exist && forceQuery != "" {
+			force, err = strconv.ParseBool(forceQuery)
+			if err != nil {
+				c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrStrconvParseBool.Wrap(err).FastGenWithCause().Error())
+				return
+			}
+		}
+
+		err = rc.RemoveStore(id, force)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, err.Error())
+			return
+		}
+
+		c.JSON(http.StatusOK, nil)
+	}
+}
+
+type updateStoresParams struct {
+	NodeState    string  `json:"node_state"`
+	LeaderWeight float64 `json:"leader_weight"`
+	RegionWeight float64 `json:"region_weight"`
+}
+
+// UpdateStoreByID will update the store according to the given ID.
+// @Tags stores
+// @version 2.0
+// @Summary Update a store's information.
+// @Param id path integer true "Store Id"
+// @Accept json
+// @Produce json
+// @Success 200
+// @Failure 400 {string} string "The input is invalid."
+// @Failure 404 {string} string "The store does not exist."
+// @Failure 500 {string} string "PD server failed to proceed the request."
+// @Router /stores [patch]
+func UpdateStoreByID() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		rc := c.MustGet(apiutil.ClusterKey).(*cluster.RaftCluster)
+		idParam := c.Param(StoreIDParamKey)
+		id, err := strconv.ParseUint(idParam, 10, 64)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrStrconvParseUint.Wrap(err).FastGenWithCause().Error())
+			return
+		}
+
+		var p updateStoresParams
+		if err := c.BindJSON(&p); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+			return
+		}
+
+		if p.LeaderWeight < 0 || p.RegionWeight < 0 {
+			c.AbortWithStatus(http.StatusBadRequest)
+			return
+		}
+
+		if err := rc.SetStoreWeight(id, p.LeaderWeight, p.RegionWeight); err != nil {
+			if errors.Is(err, errs.ErrStoreNotFound) {
+				c.AbortWithStatus(http.StatusNotFound)
+				return
+			}
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		if p.NodeState != "" {
+			switch p.NodeState {
+			case metapb.NodeState_Serving.String():
+				if err = rc.UpStore(id); errors.Is(err, errs.ErrStoreNotFound) {
```

Comment:
So If we meet an error here, the `update` operator of  `store-weight`  was success, while the request returns failed?